packages/core/src/lifecycle.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * Action to handle life cycle events
+ */
+export interface LifeCycleAction {
+  start(): Promise<void> | void;
+  stop(): Promise<void> | void;
```

Comment:
Let's discuss this API. Shouldn't we provide some kind of a context in which the event is happening? Without a context, how can a life-cycle-action handler access relevant data like the application and/or the server being started?

Reply:
If we switch to `observer` pattern, yes. For simple `start/stop`, I don't see the need to have an argument for context. Other information can be injected from the application context.